### PR TITLE
add foxi/cci.20210217

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

another mandatory dependency of libtorch (the last one I hope)

closes https://github.com/conan-io/conan-center-index/issues/4946

### CODE REVIEW

**Review by danimtb — APPROVED** [review]

**Review by jgsogo — COMMENTED** [review]
Is it worth sending the patch upstream? The original sources don't compile without this change... I'm really surprised not to see an issue about it in the repository.

**Comment by SpaceIm** [comment]
Yes I can submit a PR (two actually, another for the `tools.replace_in_file()` in CMakeLists).

**Review by memsharded — COMMENTED** [review]
Recipe is good, if upstream patches are not accepted for some reason, could be merged.
Maybe also worth asking them to tag a release, so we could use proper version.

**Comment by SpaceIm** [comment]
With this last recipe, I can submit a recipe for `libtorch`, to burn CCI infrastructure.
It's ready. Well at least it builds and can be consumed with CPU support on Windows and Macos, shared and static. And I'm testing compilation with CUDA.
There are few things to improve for several options, but the current state is not too bad.

File: recipes/foxi/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    name = "foxi"
+    description = "ONNXIFI with Facebook Extension."
+    license = "MIT"
+    topics = ("conan", "foxi", "onnxifi")
+    homepage = "https://github.com/houseroad/foxi"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"fPIC": [True, False]}
+    default_options = {"fPIC": True}
```
Comment (prince-chrismc) [inline recipes/foxi/all/conanfile.py:17]: Is it possible to build shared?

Hmmm nvm its not supported https://github.com/houseroad/foxi/blob/bd6feb6d0d3fc903df42b4feb82a602a5fcb1fd5/CMakeLists.txt#L52

File: recipes/foxi/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        tools.replace_in_file(cmakelists,
+                              "DESTINATION lib",
+                              "RUNTIME DESTINATION bin ARCHIVE DESTINATION lib LIBRARY DESTINATION lib")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["FOXI_WERROR"] = False
+        self._cmake.configure()
```
Comment (prince-chrismc) [inline recipes/foxi/all/conanfile.py:55]: Build folder?

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/foxi/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/foxi/all/conanfile.py:55]: There is no strong reason to use out-of-source build, in-source build just works.

**Pull request merged**